core/src/data/literal.rs

```diff
@@ -172,6 +176,28 @@ impl<'a> Literal<'a> {
         }
     }
 
+    pub fn bitwise_and(&self, other: &Literal<'a>) -> Result<Literal<'static>> {
+        match (self, other) {
+            (Number(l), Number(r)) => match (l.to_i64(), r.to_i64()) {
+                (Some(l), Some(r)) => match (l & r).to_bigint() {
+                    Some(v) => Ok(Number(Cow::Owned(BigDecimal::new(v, 0)))),
+                    None => Err(LiteralError::UnreachableBinaryArithmetic.into()),
+                },
+                _ => Err(LiteralError::UnsupportedBinaryArithmetic(
```

Comment:
Can we add a new error variant that is not `UnsupportedBinaryArithmetic`?
I don't think it's appropriate since it's an unsupported error related to arithmetic.